charm/kafka/lib/charms/layer/kafka.py

```diff
@@ -13,45 +13,49 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import ipaddress
 import os
-import re
 import shutil
-import socket
 
+from subprocess import check_call
 from pathlib import Path
 
 from charmhelpers.core import hookenv, host
 from charmhelpers.core.templating import render
+from charms.reactive.relations import RelationBase
+from charms.layer import snap
 
+KAFKA_PORT = 9093
+KAFKA_SNAP = 'kafka'
+KAFKA_SERVICE = 'snap.{}.kafka.service'.format(KAFKA_SNAP)
+KAFKA_SNAP_DATA = '/var/snap/{}/common'.format(KAFKA_SNAP)
 
-KAFKA_PORT=9093
-KAFKA_SNAP='kafka'
-KAFKA_SERVICE='snap.{}.kafka.service'.format(KAFKA_SNAP)
-KAFKA_SNAP_DATA='/var/snap/{}/common'.format(KAFKA_SNAP)
 
 class Kafka(object):
     def open_ports(self):
+        '''
+        Attempts to open the Kafka port.
+        '''
         hookenv.open_port(KAFKA_PORT)
 
     def close_ports(self):
+        '''
+        Attempts to close the Kafka port.
+        '''
         hookenv.close_port(KAFKA_PORT)
 
-    def configure_kafka(self, zk_units, log_dir=None):
-        # Get ip:port data from our connected zookeepers
-        zks = []
-        for unit in zk_units:
-            ip = resolve_private_address(unit['host'])
-            zks.append("%s:%s" % (ip, unit['port']))
-        zks.sort()
-        zk_connect = ",".join(zks)
+    def install(self, zks='', log_dir=None):
+        '''
+        Generates client-ssl.properties and server.properties with the current
+        system state.
+        '''
+        zk_connect = zks or ",".join(self.read_zk_peers())
```

Comment:
Are we sure that the contents of zk_connect here will be equivalent to the above code we're replacing?